### ISSUE TITLE
Can't run alternator with both HTTP and HTTPS

### ISSUE THREAD

**Issue by dimaqq** [scylladb/scylla#10016]
If I use this flags in `docker-compose.yml`:
`command: --smp=1 --skip-wait-for-gossip-to-settle 0 --alternator-port=8000 --alternator-https-port=8443 --alternator-write-isolation=always`

I see an error in the container logs:

```
aerospike-test-scylladb-1  | FATAL: Exception during startup, aborting: boost::wrapexcept<boost::program_options::multiple_occurrences> (option '--alternator-address' cannot be specified more than once)
```

However, if I drop HTTPS the error goes away:
`command: --smp=1 --skip-wait-for-gossip-to-settle 0 --alternator-port=8000 --alternator-write-isolation=always`

Version info:

```command
> docker pull scylladb/scylla
Using default tag: latest
latest: Pulling from scylladb/scylla
Digest: sha256:b89abe41ed96dac16dc2c9ac624d947e0f24dd7300d5de095d0cbeb847a8b437
Status: Image is up to date for scylladb/scylla:latest
docker.io/scylladb/scylla:latest

> docker images | ag scy
scylladb/scylla                                          latest                 318063e779bc   5 weeks ago    809MB
```

**Comment by nyh** [comment]
This is not a bug in Scylla itself, but rather in our Docker setup script `dist/docker/scyllasetup.py`:
```python
        if self._alternatorPort is not None:
            args += ["--alternator-address %s" % self._alternatorAddress]
            args += ["--alternator-port %s" % self._alternatorPort]

        if self._alternatorHttpsPort is not None:
            args += ["--alternator-address %s" % self._alternatorAddress]
            args += ["--alternator-https-port %s" % self._alternatorHttpsPort]
```

The `--alternator-address` option should be provided only once, not twice.
I'll prepare a patch.

By the way, it is unfortunate we have so many argument translation layers in the docker script (we have this `scyllasetup.py`, and we have `commandlineparser.py`). The docker script should have just passed these options, untranslated, to Scylla. See also #4625.

**Comment by nyh** [comment]
I sent a patch for this issue to the mailing list, titled `docker: don't repeat "--alternator-address" option twice`.
The frustrating thing is that although that patch fixes the "cannot be specified more than once" message,  if I try to run with SSL, even just ssl:
```
podman run --name scylla -d -p 8000:8000 -p 8443:8443 oci-archive:build/dev/dist/docker/scylla-5.0.dev-0.20220202.f285d8873 --alternator-https-port=8443 --alternator-write-isolation=always
```
Scylla fails to boot:
```
scylla: ./seastar/include/seastar/core/sharded.hh:563: seastar::sharded<alternator::server>::~sharded() [Service = alternator::server]: Assertion `_instances.empty()' failed.
```
I don't know if this is a regression or what. If I try to run exactly the same command line outside docker, it works. Also I thought I fixed that "assertion instances empty" bug (commit 79776ff2ffab287180cd7baa397668b52f96256d) so I don't understand how it came back :-(

**Comment by nyh** [comment]
> Scylla fails to boot:
> 
> ```
> scylla: ./seastar/include/seastar/core/sharded.hh:563: seastar::sharded<alternator::server>::~sharded() [Service = alternator::server]: Assertion `_instances.empty()' failed.
> ```
> 

I can reproduce this even in the official docker image. I'll open a new issue about that.

**Comment by avikivity** [comment]
Backported to 4.4, 4.5, 4.6. 5.0 had the fix already.

**Comment by nyh** [comment]
@avikivity removed the backport candidate tag from this issue in February after backporting, somehow it got re-added for no reason in July. I have no idea why... Removing the tag again.
